contrib/platform/test/com/sun/jna/platform/win32/PsapiUtilTest.java

```diff
@@ -0,0 +1,77 @@
+/* Copyright (c) 2020 Torbjörn Svensson, All Rights Reserved
+ *
+ * The contents of this file is dual-licensed under 2
+ * alternative Open Source/Free licenses: LGPL 2.1 or later and
+ * Apache License 2.0. (starting with JNA version 4.0.0).
+ *
+ * You can freely decide which license you want to apply to
+ * the project.
+ *
+ * You may obtain a copy of the LGPL License at:
+ *
+ * http://www.gnu.org/licenses/licenses.html
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "LGPL2.1".
+ *
+ * You may obtain a copy of the Apache License at:
+ *
+ * http://www.apache.org/licenses/
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+package com.sun.jna.platform.win32;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import org.junit.Test;
+
+import com.sun.jna.platform.win32.WinNT.HANDLE;
+
+/**
+ * Applies API tests on {@link PsapiUtil}.
+ *
+ * @author Torbj&ouml;rn Svensson, azoff[at]svenskalinuxforeninen.se
+ */
+@SuppressWarnings("nls")
+public class PsapiUtilTest {
+    @Test
+    public void enumProcesses() {
+        int[] pids = PsapiUtil.enumProcesses();
+
+        assertTrue("List should contain atleast one pid", pids.length > 0);
+
+        int myPid = Kernel32.INSTANCE.GetCurrentProcessId();
+        boolean foundMyPid = false;
+        for (int i = 0; i < pids.length; i++) {
+            if (pids[i] == myPid) {
+                foundMyPid = true;
+                break;
+            }
+        }
+        assertTrue("List should contain my pid", foundMyPid);
+    }
+
+    @Test
+    public void getProcessName() {
+        String name = PsapiUtil.getProcessName(Kernel32.INSTANCE.GetCurrentProcessId());
+        assertNotNull("Name for pid should not be null", name);
+        assertTrue("Name for pid should contain java", name.contains("java"));
+
+        HANDLE hProcess = Kernel32.INSTANCE.GetCurrentProcess(); // pseudo handle, no need to close
+        name = PsapiUtil.getProcessName(hProcess);
+        assertNotNull("Name for handle should not be null", name);
+        assertTrue("Name for handle should contain java", name.contains("java"));
+
+        try {
+            PsapiUtil.getProcessName(0); // the system process
+            fail("Should never reach here");
+        } catch (Win32Exception e) {
```

Comment:
Generally in test cases where an exception is intentionally thrown for the test, I like to name the exception `expected`.